### PR TITLE
Mark ZERO_TOLERANCE as deprecated

### PR DESCRIPTION
**Pull request by asmaloney**
Part of #20

### CODE REVIEW

File: include/ManualSegmentationTools.h [hunk 1]
```diff
@@ -112,7 +112,7 @@ namespace CCCoreLib
 				: insideMesh(nullptr)
 				, outsideMesh(nullptr)
 				, generateOutsideMesh(false)
-				, epsilon(ZERO_TOLERANCE)
+				, epsilon(std::numeric_limits<double>::epsilon())
```
[the remark this answers is not in the document]
Comment (dgirardeau) [reply include/ManualSegmentationTools.h:115]: And I realize now how having GreaterThanEpsilon(float) and GreaterThanEpsilon(double) defined differently is dangerous here... We might have to be more explicit.

File: src/WeibullDistribution.cpp [hunk 2]
```diff
@@ -289,8 +290,8 @@ double WeibullDistribution::ComputeG(const ScalarContainer& values, double r, Sc
 
 	if (zeroValues)
 	{
-		double ln_v = log(ZERO_TOLERANCE) * zeroValues;
-		double v_a = pow(ZERO_TOLERANCE / valueRange, static_cast<double>(r));
+		const double ln_v = log(std::numeric_limits<double>::epsilon()) * zeroValues;
```
Comment (dgirardeau) [inline src/WeibullDistribution.cpp:293]: Same as above

**Comment by asmaloney** [comment]
Oh, alright! So I think when we discussed this before we talked about adding a float version of the var.

So according to your comment, that would look like this:

```cpp
constexpr double ZERO_TOLERANCE_D = static_cast<double>(std::numeric_limits<float>::epsilon());
constexpr float ZERO_TOLERANCE_F = std::numeric_limits<float>::epsilon();
```

Then we can at least avoid extra casting and it's more explicit.

Would we then use `ZERO_TOLERANCE_D` in the double versions of _LessThanEpsilon_, etc.? Or are there cases where we actually want the real double epsilon?

**Comment by asmaloney** [comment]
> Or are there cases where we actually want the real double epsilon?

Actually, given that we changing where `ZERO_TOLERANCE` is used - there shouldn't be!

**Comment by dgirardeau** [comment]
Yes I was going to say that there should only be one 'value' for ZERO_TOLERANCE (in two flavors). And if we need another epsilon value, then we would use something else than ZERO_TOLERANCE.

**Comment by dgirardeau** [comment]
(or is flavours? :D )

**Comment by asmaloney** [comment]
Is this a reasonable explanation for the comments?

> ZERO_TOLERANCE_F and ZERO_TOLERANCE_D are used when we compare, or want to set a variable to, "close to zero". It is defined as std::numeric_limits<float>::epsilon() for both single and double precision numbers because using std::numeric_limits<double>::epsilon() results in numbers that are too small for our purposes.

**Comment by dgirardeau** [comment]
I guess it's std::numeric_limits'<'float'>'::epsilon() but markdown is removing the '<'float'>' or '<'double'>' ? :D.

Then yes.

**Comment by asmaloney** [comment]
Ah yes :-)

File: include/CCConst.h [hunk 3]
```diff
@@ -34,8 +33,16 @@ namespace CCCoreLib
...
-	[[deprecated( "Use std::numeric_limits<>::epsilon or CCCoreLib::LessThanEpsilon()/CCCoreLib::GreaterThanEpsilon()" )]]
-	constexpr double ZERO_TOLERANCE = static_cast<double>(FLT_EPSILON);
+	[[deprecated( "Use ZERO_TOLERANCE_F/ZERO_TOLERANCE_D or CCCoreLib::LessThanEpsilon()/CCCoreLib::GreaterThanEpsilon()" )]]
+	constexpr double ZERO_TOLERANCE = std::numeric_limits<float>::epsilon();
+	
+	//! ZERO_TOLERANCE_F is used to set a float variable to "close to zero".
+	constexpr float ZERO_TOLERANCE_F = std::numeric_limits<float>::epsilon();
+	
+	//! ZERO_TOLERANCE_D is used to set a double variable to "close to zero".
+	//! It is defined as std::numeric_limits::epsilon() because using
```
Comment (dgirardeau) [inline include/CCConst.h:43]: Is github also hiding the '<'float'>' or '<'double'>' in the comments?!

Comment (asmaloney) [reply include/CCConst.h:43]: lol - no - I copy-pasted the comment from above :-)

**Pull request merged**